search-app/src/cmr/search/site/data.clj

```diff
@@ -76,6 +76,19 @@
   [results]
   (map :concept-id (:items results)))
 
+(defn-timed get-granule-count
+  "Get the granule count of a collection concept id given some query parameters"
+  [context collection-concept-id query-params]
+  (let [query (query-model/query {:concept-type :granule
+                                  :condition (gc/and-conds (query-svc/generate-query-conditions-for-parameters
+                                                            context
+                                                            :granule
+                                                            (merge {:collection-concept-id collection-concept-id} query-params)))
+                                  :page-size :unlimited
+                                  :result-format :query-specified
+                                  :result-fields [:collection-concept-id]})]
+    (count (:items (query-exec/execute-query context query)))))
```

Comment:
Page-size unlimited could result in long query times. You should be able to run the query with page-size of 0 and ES should give you the total count that is possible with the given query.

Reply:
Agree with Jay that you should use page-size 0 here. 
On a higher level, you should not search for granule counts one collection at a time. Take a look at my other comment on using aggregation instead.

Reply:
done